packages/Interface/src/DTK_C_API.cpp

```diff
@@ -349,98 +363,135 @@ bool DTK_is_initialized() { return DataTransferKit::isInitialized(); }
 
 void DTK_finalize() { DataTransferKit::finalize(); }
 
-void DTK_set_function( DTK_UserApplicationHandle handle, DTK_FunctionType type,
-                       void ( *f )(), void *user_data )
+int DTK_set_function( DTK_UserApplicationHandle handle, DTK_FunctionType type,
```

Comment:
No. The return codes and `errno` play together. I think the idea is that first you get some indication that the function failed (like negative return, 0 pointer, or something like that), and that query `errno` for the reason why it failed.